lib/auth.js

```diff
@@ -6,10 +6,59 @@ const _ = require('lodash'),
   references = require('./services/references'),
   session = require('express-session'),
   flash = require('express-flash'),
+  responses = require('./responses'),
   handlebars = require('handlebars'),
   basicAuth = require('basic-auth'),
   fs = require('fs'),
-  path = require('path');
+  path = require('path'),
+  AUTH_LEVELS_MAP = {
+    ADMIN: 'admin',
+    WRITE: 'write',
+    READ: 'read'
+  },
+  AUTH_LEVEL_VALUES = {
+    admin: 0,
+    write: 1,
+    read: 2
```

Comment:
looks good for now. If we need permissions that are not strictly-ordered in future it would be fun to use bit masks